terraform/modules/front_door/waf.tf

```diff
@@ -0,0 +1,164 @@
+provider "aws" {
+  alias  = "north-virginia"
+  region = "us-east-1"
+
+  assume_role {
+    role_arn = var.provider_role_arn
+  }
+}
+
+resource "aws_wafv2_web_acl" "this" {
+  #checkov:skip=CKV2_AWS_31:TODO CLDC-2781 setup WAF logging in cloudwatch
+  name        = var.prefix
+  description = "Web ACL to restrict traffic to CloudFront"
+  provider    = aws.north-virginia
+  scope       = "CLOUDFRONT"
+
+  default_action {
+    allow {}
+  }
+
+  visibility_config {
+    cloudwatch_metrics_enabled = true
+    metric_name                = "waf-metrics"
+    sampled_requests_enabled   = false
+  }
+
+  rule {

```

Comment:
Wasn't completely sure about rule order / priority here but generally felt this was the likely order of importance. I think as long as a bad request gets caught the priority doesn't matter too much:

- `AWSManagedRulesAmazonIpReputationList` - block IPs with a bad rep first
- `AWSManagedRulesCommonRuleSet` - block requests with common web app exploits
- `AWSManagedRulesKnownBadInputsRuleSet` - block requests with bad request patterns known for common exploits or discovery of vulnerabilities
- `AWSManagedRulesSQLiRuleSet` - block requests associated with exploitation of SQL databases
- `AWSManagedRulesLinuxRuleSet` - block requests associated with exploits specific to LINUX systems
- `AWSManagedRulesUnixRuleSet` - block requests associated with exploits specific to POSIX systems (AWS suggests this should be used with the Linux rule set)

More info about these can be found [here](https://docs.aws.amazon.com/waf/latest/developerguide/aws-managed-rule-groups-list.html). See the `baseline`, `use-case specific` and `IP reputation` sections